cdk/src/open-data-platform/data-plane/data-import/write-water-systems-data-handler.handler.ts

```diff
@@ -194,11 +215,20 @@ export async function handler(_: APIGatewayProxyEvent): Promise<APIGatewayProxyR
     // See https://docs.aws.amazon.com/rds/index.html.
     const rdsService = new AWS.RDSDataService();
 
+    try {
+      const rowsInDb = (await determineRowsWritten(rdsService)).records[0][0].longValue;
+      startIndex = Math.max(parseInt(rowsInDb) - NUMBER_ROWS_BUMPER, startIndex);
```

Comment:
Same comment as before. I'm 99% sure you can't assume that N rows in the DB are the first N rows in the file. Is that acceptable?

Reply:
If you only pass in numberOfRows that you know it can handle and it does not time out, it will do the first numberOfRows to write in the file, right? It doesn't matter if 800-900 get committed before 900-1000 as long as 10k rows go in

Reply:
SGTM

Reply:
Cool resolving and submitting